imager_images/tests.py

```diff
@@ -1,33 +1,473 @@
+
+    def test_published_status_in_context(self):
+        """Prove that published status in context."""
+        self.assertIn('status', self.response.context)
+
+
+class AlbumsViewTestCase(SetupTestCase):
+    """Define class to test albums view."""
+    def setUp(self):
+        """Define setup for tests."""
+        self.setUp = super(AlbumsViewTestCase, self).setUp()
+        self.url = reverse('albums', kwargs={'pk': self.album.pk})
+        self.response = self.client.get(self.url)
+        self.template = 'imager_images/album_page.html'
+
+    def test_auth_user_has_access_to_albums(self):
+        """Prove that response code is 200 for auth users."""
+        self.assertEquals(self.response.status_code, 200)
+
+    def test_right_template_is_used(self):
+        """Prove that right template is used to render album page."""
+        self.assertTemplateUsed(self.response, self.template)
+
+    def test_album_attributes_show_up(self):
+        """Prove that all attributes show up on the page."""
+        attr = [
+            self.album.title,
+            self.album.description,
+            self.album.photos.all()[0].photo.url,
+            self.album.photos.all()[0].is_cover,
+            ]
+        for x in attr:
+            self.assertContains(self.response, x)
+
+    def test_published_status_in_context(self):
+        """Prove that published status in context."""
+        self.assertIn('status', self.response.context)
+
+
+class LibraryViewTestCase(SetupTestCase):
+    """Define test class for libarry view."""
+    def setUp(self):
+        """Define setup for tests."""
+        self.setUp = super(LibraryViewTestCase, self).setUp()
+        self.url = reverse('library')
+        self.response = self.client.get(self.url)
+        self.template = 'imager_images/library_page.html'
+
+    def test_auth_user_has_access_to_library(self):
+        """Prove that response code is 200 for auth users."""
+        self.assertEquals(self.response.status_code, 200)
+
+    def test_right_template_is_used(self):
+        """Prove that right template is used to render library page."""
+        self.assertTemplateUsed(self.response, self.template)
+
+    def test_buttons_present(self):
+        """Prove that expected buttons are present."""
+        buttons = [
+            reverse('album_add'),
+            reverse('photo_add'),
+            reverse('album_edit', kwargs={'pk': self.album.pk}),
+            reverse('photo_edit', kwargs={'pk': self.photo1.pk}),
+            reverse('photo_edit', kwargs={'pk': self.photo2.pk})
+        ]
+        for x in buttons:
+            expected = 'href="{}"'.format(x)
+            self.assertContains(self.response, expected)
+
+    def test_albums_and_photos_in_context(self):
+        """Prove that 'albums' and 'photos' are in response context."""
+        list_ = ['albums', 'photos']
+        for x in list_:
+            self.assertIn(x, self.response.context)
```

Comment:
Should probably test that the number of photos that you expect to be on the page are there, as well as the number of albums.